group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,523 @@
+        topics.put(topic2Uuid, new AssignmentTopicMetadata(2));
+
+        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, Collections.singleton(0));
+        currentAssignmentForA.put(topic2Uuid, Collections.singleton(0));
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForA
+        ));
+
+        Map<Uuid, Set<Integer>> currentAssignmentForB = new HashMap<>();
+        currentAssignmentForB.put(topic1Uuid, Collections.singleton(1));
+        currentAssignmentForB.put(topic2Uuid, Collections.singleton(1));
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForB
+        ));
+
+        // Add a new consumer to trigger a re-assignment
+        members.put(consumerC, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            Collections.emptyMap()
+        ));
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<String, Map<Uuid, Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        expectedAssignment.put(consumerA, mkAssignment(mkTopicAssignment(topic1Uuid, 0)));
+        expectedAssignment.put(consumerB, mkAssignment(mkTopicAssignment(topic1Uuid, 1)));
+        // Topic 2 Partitions Assignment
+        expectedAssignment.get(consumerA).putAll(mkAssignment(mkTopicAssignment(topic2Uuid, 0)));
+        expectedAssignment.get(consumerB).putAll(mkAssignment(mkTopicAssignment(topic2Uuid, 1)));
+
+        // Consumer C shouldn't get any assignment, due to stickiness A, B retain their assignments
+        assertNull(computedAssignment.members().get(consumerC));
+        assertAssignment(expectedAssignment, computedAssignment);
+        assertCoPartitionJoinProperty(computedAssignment);
+    }
+
+    @Test
+    public void testReassignmentWhenOnePartitionAddedForTwoConsumersTwoTopics() {
+        // Simulating adding a partition - originally T1 -> 3 Partitions and T2 -> 3 Partitions
+        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        topics.put(topic1Uuid, new AssignmentTopicMetadata(4));
+        topics.put(topic2Uuid, new AssignmentTopicMetadata(4));
+
+        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, new HashSet<>(Arrays.asList(0, 1)));
```

Comment:
nit: We have `mkSet` in `org.apache.kafka.common.utils`. You can statically import it and use it.